rfd/0095-application-access-okta-integration.md

```diff
@@ -0,0 +1,315 @@
+}
+```
+
+### Okta configuration
+
+The Okta service should have its own unique top level configuration. The following config fields will
+be available in the Teleport config YAML, along with associated environment variables:
+
+| Name | Environment Variable | Required | Description |
+|------|----------------------|----------|-------------|
+| `api_url` | `TELEPORT_OKTA_API_URL` | :heavy_check_mark: | The API URL so that Teleport knows which Okta endpoint to hit.
+| `role_prefix` | `TELEPORT_OKTA_ROLE_PREFIX` | :x: | A role prefix to prepend to each group role created in Teleport. Defaults to `okta`.
+| `role_template` | `TELEPORT_OKTA_ROLE_TEMPLATE` | :x: | A role template for Teleport roles created by the Okta service. Defaults to allowing application read access for applications that belong to a specific Okta group.
+
+The Okta API token must be configured using an environment variable or command line
+argument to prevent the possibility of users accidentally checking in this key.
+
+**Note**: The details of the `role_prefix` and `role_template` will be discussed further in [the Okta to Teleport mapping section](#okta-to-teleport-mappings)
+
+#### YAML example
+
+```yaml
+okta_service:
+  enabled: true
+  api_url: https://my-okta-endpoint.okta.com
+  role_prefix: okta # This is the default role prefix.
+```
+
+#### CLI exmples (API token)
+```bash
+# Command line argument example
+teleport start --okta-api-token 12345678
+
+# Environment variable example
+TELEPORT_OKTA_API_TOKEN="12345678" teleport start
+```
+
+### Okta user traits
+
+Okta will optionally use an `okta_user_id` trait to map Teleport users to Okta users if necessary.
+
+```yaml
+kind: user
+version: v2
+metadata:
+  name: alice
+spec:
+  roles: ['devs']
+  traits:
+    logins: ['admin']
+    kubernetes_groups: ['edit']
+    okta_user_id: ['name@mydomain.com']
+```
+
+If Okta is used for logging into Teleport or the Teleport user's username is the same as the
+e-mail used for Okta, this will be unnecessary.
+
+### Background synchronization
+
+The background synchronization process, which synchronizes all applications from Okta, is expected
+to run roughly every 2 minutes. This process wil translate Okta users, groups, and applications
+into their Teleport equivalents.
+
+```mermaid
+sequenceDiagram
+  participant OktaSvc as Okta Service
+  participant Okta as Okta API
+  participant AppSvc as Application Access Service
+
+  loop Every 2 minutes
+    OktaSvc->>Okta: Get list of applications
+    OktaSvc->>AppSvc: Synchronize applications in Teleport
+
+    OktaSvc->>Okta: Get list of groups
+
+    OktaSvc->>AppSvc: Synchronize groups in Teleport
+  end
+```
+
+#### Okta to Teleport mappings
+
+Okta groups and applications will be mapped by the background synchronization into Teleport roles and
+applications respectively.
+
+##### Groups
+
+A Teleport role will be created for each Okta group. Each group will be prepended with a configurable
```

Comment:
Didn't we want to create a new Teleport resource to represent Okta groups? What's the benefit of creating a role per group?

Reply:
The benefit I envisioned here was being able to use Teleport's RBAC natively to determine visibility, but I believe you've got another suggestion w/r/t this later in the doc.